### PR TITLE
Apply "instanceof pattern matching" and minor refactorings in `spring-context`

### PR DESCRIPTION
**Pull request by Drezir**

### CODE REVIEW

File: spring-context/src/main/java/org/springframework/context/annotation/BeanMethod.java [hunk 1]
```diff
@@ -56,8 +56,8 @@ public void validate(ProblemReporter problemReporter) {
 
 	@Override
 	public boolean equals(@Nullable Object obj) {
-		return ((this == obj) || ((obj instanceof BeanMethod) &&
-				this.metadata.equals(((BeanMethod) obj).metadata)));
+		return ((this == obj) || ((obj instanceof BeanMethod beanMethod) &&
+				this.metadata.equals((beanMethod).metadata)));
```
Comment (r331) [inline spring-context/src/main/java/org/springframework/context/annotation/BeanMethod.java:60]: Maybe without braces? this.metadata.equals(beanMethod.metadata)));

File: spring-context/src/main/java/org/springframework/cache/interceptor/SimpleCacheResolver.java [hunk 2]
```diff
@@ -64,7 +64,7 @@ protected Collection<String> getCacheNames(CacheOperationInvocationContext<?> co
 	 */
 	@Nullable
 	static SimpleCacheResolver of(@Nullable CacheManager cacheManager) {
-		return (cacheManager != null ? new SimpleCacheResolver(cacheManager) : null);
+		return cacheManager != null ? new SimpleCacheResolver(cacheManager) : null;
 	}
```
Comment (sbrannen) [inline spring-context/src/main/java/org/springframework/cache/interceptor/SimpleCacheResolver.java:68]: This is the standard code style in Spring Framework.

Please revert.

File: spring-context/src/main/java/org/springframework/context/annotation/BeanMethod.java (hunk 1 above)
Comment (sbrannen) [inline spring-context/src/main/java/org/springframework/context/annotation/BeanMethod.java:60]:
```suggestion
				this.metadata.equals(beanMethod.metadata)));
```

**Review by sbrannen — CHANGES_REQUESTED** [review]
Thanks for the PR.

I've requested some changes.

If you shift the focus of this PR to "instanceof pattern matching", the work can be merged sooner.

**Comment by sbrannen** [comment]
Thanks for the PR.

Due to lack of feedback on the requested changes, I have merged parts of this PR into `main` in 8783075e40881c8ca2ee52d81c76e2f6149ace38 with further refinements in e0f60dc09d9515809963f066cdfdacebd66b0042.

**Pull request closed without merging**